docs/error-handling.md

```diff
@@ -0,0 +1,297 @@
+# Support workers - Error handling
+
+One of the main benefits of Step functions is the flexibility it gives us in dealing with error conditions. For any error we encounter we have three ways of dealing with it:
+
+* Unlimited Retry - For failures where we believe that the cause may be temporary step functions allow us to simply retry with a back off period.
+* Limited Retry - For failures where we suspect that a retry may succeed but don't want to get stuck in an infinite retry loop if it doesn't, we can specify a maximum number of retries before the task fails permanently.
+* Fallback - For permanent failures we can catch the exception and pass it to a fallback state which should then log the error and notify developers and the user that their payment has failed.
+
+This means that we now need to do a bit more analysis of the error conditions our app is likely to encounter to group them into those which can be retried and those
+which are permanent failures. The rest of this document is an attempt to do that analysis for the Monthly contributions step function
+
+## Logging
+Most errors should be logged to Sentry so they can be investigated. However transient ones such as timeouts probably don't need to be logged in Sentry and can just be logged in Cloudwatch. This is also the case for Stripe card errors.
+
+## Types of errors
+
+### General errors
+
+<table>
```

Comment:
In the future, if you want to avoid html tabes, markdown also accepts them in [this format](http://www.tablesgenerator.com/markdown_tables).